lib/bootstrapper/index.ts

```diff
@@ -28,17 +27,18 @@ export class BootstrapPgStac extends Construct {
   constructor(scope: Construct, id: string, props: BootstrapPgStacProps) {
     super(scope, id);
 
-    const { pgstacVersion = DEFAULT_PGSTAC_VERSION } = props;
     const handler = new aws_lambda.Function(this, "lambda", {
-      handler: "handler.handler",
-      runtime: aws_lambda.Runtime.PYTHON_3_8,
-      code: aws_lambda.Code.fromDockerBuild(__dirname, {
+      ...props.lambdaFunctionOptions ?? {
+        runtime: aws_lambda.Runtime.PYTHON_3_8,
+        handler: "handler.handler",
+        memorySize: 128,
+        logRetention: aws_logs.RetentionDays.ONE_WEEK,
+        timeout: Duration.minutes(2)
+      },
+      code: props.lambdaAssetCode ?? aws_lambda.Code.fromDockerBuild(__dirname, {
```

Comment:
@emileten I'm wondering if we might want to support both `AssetCode` and `DockerImageCode` for the code prop to handle cases where user asset package builds might exceed 250MB.

Reply:
@sharkinsspatial `code` in `aws_lambda.Function` must be of [type `aws_lambda.Code`](https://github.com/aws/aws-cdk/blob/5fe46d6f6ce325caa9a62ee5bb015821068e77ee/packages/aws-cdk-lib/aws-lambda/lib/code.ts#L12), and `DockerImageCode` doesn't seem to be implementing this. 

You use `DockerImageCode` with `aws_lambda.DockerImageFunction`instead. One way to do this would be, given the change suggested by Anthony : 
- detect if in `lambdaFunctionOptions` the user passed a value to `code`. 
- If there is, check its type. If it's `DockerImageCode`, then we use `DockerImageFunction`, otherwise and also if it's absent, we fall back to `Function`. 
  
So that adds an (albeit small) layer of complexity that I'd maybe rather tackle later in the absence of an immediate need.